analysers/experiments/faban.py

```diff
@@ -0,0 +1,175 @@
+        .select("experiment_id", "host", "driver_name", "stat_name", "description", "target", "result") \
+        .where("experiment_id=?", experimentID)
+    CassandraRDD.cache()
+    
+    combinations = CassandraRDD.map(lambda a: (a["host"], a["driver_name"], a["stat_name"], a["description"])).distinct().collect()
+    
+    for comb in combinations:
+        query = {}
+        
+        data = CassandraRDD.filter(lambda r: r['host'] == comb[0] and r['driver_name'] == comb[1] and r['stat_name'] == comb[2] and r['description'] == comb[3])
+        
+        target = data.first()["target"]
+        description = data.first()["description"]
+        
+        print target
+        
+        if target == "absolute":
+            mode = computeMode(data.map(lambda r: (r["result"], 1)))
+            metrics = computeMetrics(data.map(lambda r: r["result"]).collect())
+            
+            query.update({"experiment_id":experimentID, "faban_driver_host": comb[0], "faban_driver_name": comb[1], "faban_stat_name": comb[2], \
+                      "faban_stat_mode":mode[0], "faban_stat_mode_freq":mode[1], "faban_stat_description":comb[3],\
+                      "faban_stat_mean":metrics["mean"], "faban_stat_num_data_points":metrics["num_data_points"], \
+                      "faban_stat_min":metrics["min"], "faban_stat_max":metrics["max"], "faban_stat_sd":metrics["sd"], \
+                      "faban_stat_q1":metrics["q1"], "faban_stat_q2":metrics["q2"], "faban_stat_q3":metrics["q3"], "faban_stat_p95":metrics["p95"], \
+                      "faban_stat_p90":metrics["p90"], "faban_stat_p99":metrics["p99"], "faban_stat_percentiles":metrics["percentiles"], \
+                      "faban_stat_me":metrics["me"], "faban_stat_ci095_min":metrics["ci095_min"], "faban_stat_ci095_max":metrics["ci095_max"]})
+            absQueries.append(query)
+        
+        else:
+            min = data.map(lambda r: r["result"]).min()
+            max = data.map(lambda r: r["result"]).max()
+            query.update({"experiment_id":experimentID, "faban_driver_host": comb[0], "faban_driver_name": comb[1], "faban_stat_name": comb[2], \
+                          "faban_stat_description":comb[3],\
+                          "faban_stat_min":min, "faban_stat_max":max})
+            statQueries.append(query)
+            
+    return (absQueries, statQueries)
+        
+    
+def getAnalyserConf(SUTName):
+    from commons import getAnalyserConfiguration
+    return getAnalyserConfiguration(SUTName)
+
+def main():
+    # Takes arguments
+    args = json.loads(sys.argv[1])
+    experimentID = str(args["experiment_id"])
+    SUTName = str(args["sut_name"])
+    containerID = str(args["container_id"])
+    hostID = str(args["host_id"])
+    
+    # Set configuration for spark context
+    conf = SparkConf().setAppName("Faban analyser")
+    sc = CassandraSparkContext(conf=conf)
+    
+    analyserConf = getAnalyserConf(SUTName)
+    cassandraKeyspace = "benchflow"
+    
```

Comment:
@VincenzoFerme Since we use Spark so heavily using multiple threads doesn't matter, since when the Spark context is called it will not do any extra parallelization.